pallets/settlement/src/lib.rs

```diff
@@ -1075,11 +1085,17 @@ impl<T: Trait> Module<T> {
             instruction_details.status == InstructionStatus::Pending,
             Error::<T>::InstructionNotPending
         );
-        if let Some(valid_from) = instruction_details.valid_from {
+        if let Some(trade_date) = instruction_details.trade_date {
             ensure!(
-                <pallet_timestamp::Module<T>>::get() >= valid_from,
-                Error::<T>::InstructionWaitingValidity
+                <pallet_timestamp::Module<T>>::get() >= trade_date,
+                Error::<T>::InstructionDatesInvalid
             );
```

Comment:
I'd suggest removing this check to allow issuers to record trades that have already happened offchain.